periodic_group.go

```diff
@@ -0,0 +1,112 @@
+package amboy
+
+import (
+	"context"
+	"time"
+
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/mongodb/grip/recovery"
+	"github.com/pkg/errors"
+)
+
+// GroupQueueOperation describes a single queue population operation
+// for a group queue.
+type GroupQueueOperation struct {
+	Operation QueueOperation
+	Queue     string
+	Check     func(context.Context) bool
+}
+
+// IntervalGroupQueueOperation schedules jobs on a queue group with
+// similar semantics as IntervalQueueOperation.
+//
+// Operations will continue to run as long as the context is not
+// canceled. If you do not pass any GroupQueueOperation items to this
+// function, it panics.
+func IntervalGroupQueueOperation(ctx context.Context, qg QueueGroup, interval time.Duration, startAt time.Time, conf QueueOperationConfig, ops ...GroupQueueOperation) {
+	if len(ops) == 0 {
+		panic("queue group operation must contain operations")
+	}
+
+	go func() {
+		var err error
+
+		if interval <= time.Microsecond {
+			grip.Criticalf("invalid interval queue operation '%s'", interval)
+			return
+		}
+
+		defer func() {
+			err = recovery.HandlePanicWithError(recover(), err, "interval background job scheduler")
+
+			if err != nil {
+				if !conf.ContinueOnError {
+					return
+				}
+
+				if ctx.Err() != nil {
+					return
+				}
+
+				IntervalGroupQueueOperation(ctx, qg, interval, startAt, conf, ops...)
+			}
+		}()
+
+		waitUntilInterval(ctx, startAt, interval)
+
+		ticker := time.NewTicker(interval)
+		defer ticker.Stop()
+
+		if ctx.Err() != nil {
+			return
+		}
+		count := 1
```

Comment:
nit: can we move this count to the for loop below this where it's actually used?